.github/workflows/release.yml

```diff
@@ -18,17 +18,12 @@ jobs:
         with:
           persist-credentials: false
 
-      - name: Export tag
-        id: vars
-        run: echo tag=${GITHUB_REF#refs/*/} >> $GITHUB_OUTPUT
-        if: ${{ github.event_name == 'release' }}
-
       - name: Update project version
+        if: ${{ github.event_name == 'release' }}
         run: |
           sed -i "s/^version = \".*\"/version = \"$RELEASE_VERSION\"/" pyproject.toml
         env:
-          RELEASE_VERSION: ${{ steps.vars.outputs.tag }}
-        if: ${{ github.event_name == 'release' }}
+          RELEASE_VERSION: ${{ github.ref_name }}
```

Comment:
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#github-context

This basically takes the tag name in our usage, taken from `refs/tags/<tag>`.